### PR TITLE
Additional testing and incremental refactor

### PR DESCRIPTION
**Pull request by kthomas**
This PR provides testing of the deployment of each of the workload types.

Some additional incremental refactoring can also be expected.

Closes https://github.com/synadia-io/nex/issues/40.

### CODE REVIEW

File: spec/spec_suite_test.go [hunk 1]
```diff
@@ -99,9 +99,34 @@ func startNATS(storeDir string) (*server.Server, *nats.Conn, *int, error) {
...
+		Storage:     nats.MemoryStorage,
+	})
+	if err != nil {
+		return nil, nil, nil, fmt.Errorf("failed to create jetstream object store: %s", err)
+	}
+
 	return ns, nc, &port, nil
 }
 
+// return a new NATS connection to the server running on `_fixtures.natsPort`
```
[the remark this answers is not in the document]
Comment (kthomas) [reply spec/spec_suite_test.go]: Removed 👍🏻

File: internal/node/config.go [hunk 2]
```diff
@@ -49,6 +48,10 @@ type NodeConfiguration struct {
 func (c *NodeConfiguration) Validate() bool {
 	c.Errors = make([]error, 0)
 
+	if c.MachinePoolSize < 1 {
```
Comment (jordan-rash) [inline internal/node/config.go:51]: instead of erroring here, can we warn and set it to 1?

Comment (kthomas) [reply internal/node/config.go:51]: Not ideal IMHO, means you can have a malformed config and get away with it.

Comment (kthomas) [reply internal/node/config.go:51]: Error gives the user a chance to fix their faulty config..

File: internal/node/machine_mgr.go [hunk 3]
```diff
@@ -112,31 +113,40 @@ func (m *MachineManager) Start() {
...
+				continue
+			}
+
 			vm, err := createAndStartVM(m.ctx, m.config, m.log)
 			if err != nil {
 				m.log.Warn("Failed to create VMM for warming pool.", slog.Any("err", err))
 				continue
 			}
 
 			go m.awaitHandshake(vm.vmmID)
```
Comment (jordan-rash) [inline internal/node/machine_mgr.go:143]: if this go routine is non-blocking, the next line adds the VM to the map without verifying a handshake. Is that the behavior we want?

Comment (kthomas) [reply internal/node/machine_mgr.go:143]: Yes because we need to be able to find it later to shut it down, even if it failed the handshake.

**Review by jordan-rash — COMMENTED** [review]
What happens if I start two nodes next to each other and the second one doesn't have "preserve network"?

File: internal/node/config.go (hunk 2 above)
Comment (jordan-rash) [reply internal/node/config.go:51]: meh, okay

**Comment by kthomas** [comment]
> What happens if I start two nodes next to each other and the second one doesn't have "preserve network"?

Surely something bad would happen. You could accidentally do that now by starting a node and manually purging CNI. We need to handle moving to a new subnet when the first is fully utilized. We can consider this edge case at that time as well.

**Review by jordan-rash — APPROVED** [review]
Amazing

**Pull request merged**